client/http/client.go

```diff
@@ -147,33 +117,18 @@ func (ci *clientInner) requestWithRetry(
 	headerOpts ...HeaderOption,
 ) error {
 	var (
-		statusCode             int
-		err                    error
-		addr                   string
-		pdAddrs, leaderAddrIdx = ci.getPDAddrs()
+		err        error
+		statusCode int
 	)
-	// Try to send the request to the PD leader first.
-	if leaderAddrIdx != -1 {
-		addr = pdAddrs[leaderAddrIdx]
+	clients := ci.sd.GetAllServiceClients()
+	for _, cli := range clients {
```

Comment:
Can we maintain the original logic of requesting the leader first and then the followers? This is to reduce the unnecessary redirect as much as possible.